### ISSUE TITLE
[Bug] CollectionView Items are reversed on AddRange

### ISSUE THREAD

**Issue by YZahringer** [xamarin/Xamarin.Forms#10826]
### Description
Using the `AddRange` method to optimize the insert of elements in an `ObservableCollection` binded to `CollectionView` adds them in reverse order.

I've tried with two different implementations of `Range ObservableCollection`, same issue:
- [MVVM Helpers](https://github.com/jamesmontemagno/mvvm-helpers/)
- [RangeObservableCollection](https://gist.github.com/weitzhandler/65ac9113e31d12e697cb58cd92601091)

### Expected Behavior
Items are added in order with `AddRange`.

### Actual Behavior
Items are added and removed in reverse order.

### Basic Information

- Version with issue: 4.6.0.800
- Last known good version: -
- IDE: Visual Studio 2019 16.6.0
- Platform Target Frameworks: 
  - iOS: (only tested on UWP)
  - Android: (only tested on UWP)
  - UWP:  16299

#### With multiple `Add` on `ObservableCollection`
![2zP8mzuwKk](https://user-images.githubusercontent.com/4254116/82688367-0f3e8780-9c59-11ea-8692-e72e82f9e260.gif)

#### With `AddRange` on `ObservableCollection`
![HR3C7AUvav](https://user-images.githubusercontent.com/4254116/82688438-2f6e4680-9c59-11ea-84e4-bf6e1a284f1e.gif)
(items are added in reverse order and remove it in reverse order as well)

### Reproduction Link
[CollectionView AddRange sample](https://github.com/MADSENSE/Madsense.XamarinForms.Sample/tree/collectionView-AddRange)

**Comment by jsuarezruiz** [comment]
@YZahringer Thanks for the reproduction sample.  I have done several tests and can reproduce the issue using 4.6 and 4.7-pre2.

<img width="1202" alt="Captura de pantalla 2020-06-02 a las 13 55 36" src="https://user-images.githubusercontent.com/6755973/83517315-f8a9f300-a4d8-11ea-8be4-135c3b80efd5.png">

Then, I have created a tests in Xamarin.Forms Core Gallery using the MvvmHelpers `ObservableRangeCollection` code and, cannot reproduce the issue:
https://gist.github.com/jsuarezruiz/bbed609b8382a72c6304c4244f48b964
<img width="1237" alt="Captura de pantalla 2020-06-02 a las 13 56 16" src="https://user-images.githubusercontent.com/6755973/83517309-f6479900-a4d8-11ea-99c8-c758b4893eb2.png">
I will check if it has been fixed by another issue fix, etc.

**Comment by YZahringer** [comment]
@jsuarezruiz Thank you for the answer. In the test there is no `ItemTemplate` defined, perhaps without it does not reproduce the problem?

**Comment by jsuarezruiz** [comment]
You are right. Using strings I didn't notice the `ItemTemplate` in your example.

**Comment by MSiccDev** [comment]
Any news on this? We’re using ObservableRangeCollection for a reason...

**Comment by banjahman** [comment]
any plans on when this will get fixed? This is a pretty critical blocker for me at the moment.

**Comment by YZahringer** [comment]
Repro project updated to XF 4.8.0.1269 (issue persist)

**Comment by YZahringer** [comment]
Repro project updated to `XF 5.0.0.1829-pre6` (issue persist).

I also added and tried with new [XCT ObservableRangeCollection](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/ObjectModel/ObservableRangeCollection.shared.cs), same problem.

**Comment by amaneshi** [comment]
Also interested in this issue progress. Use latest `XF 5.0.0.2012` and reverse still there (actually on refresh, initial AddRange work normal).

**Comment by YZahringer** [comment]
.NET MAUI related issue [#6471](https://github.com/dotnet/maui/issues/6471)
